### PR TITLE
feat: allow scopes in self-signed jwt

### PR DESCRIPTION
**Pull request by bshaffer**
Depends on https://github.com/googleapis/google-auth-library-ruby/pull/324

Removes check that the user supplied scopes, as these can now be signed into the self-signed JWTs

### CODE REVIEW

**Comment by dazuma** [comment]
The code looks fine. We'll need to update the embedded sample clients to get the tests to pass (and to do that, we'll need one of the other open PRs to get merged because they have some fixes), but we should get that done within the next day or so. Thanks Brent!

**Comment by dazuma** [comment]
Looking at this again now, I realize this will _strictly_ depend on the corresponding edit to the auth library. (If this change is executed along with an old library, it will use self-signed JWT without setting the scopes properly, which could break users.) Thus, we will need to complete the auth library edit, release a new version, and then couple this change with a dependency update on the new auth library version.

**Comment by dazuma** [comment]
Do not merge until we release googleauth 0.17. Also, this PR should be coupled to a change that updates the generator to require that generated clients use googleauth 0.17 or later.

**Comment by dazuma** [comment]
Unblocked now. googleauth 0.17.0 and gapic-common 0.7.0 are released.

File: gapic-generator/templates/default/service/client/_client.erb [hunk 1]
```diff
@@ -94,8 +94,7 @@ class <%= service.client_name %>
     <%- unless service.generic_endpoint? -%>
     # Use self-signed JWT if the scope and endpoint are unchanged from default,
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/templates/default/service/client/_client.erb:95]: The change in the code was not reflected in the comments directly above.

**Review by viacheslav-rostovtsev — CHANGES_REQUESTED** [review]

File: gapic-generator/templates/default/service/client/_client.erb (hunk 1 above)
Comment (dazuma) [reply gapic-generator/templates/default/service/client/_client.erb:95]: Good catch on that. @bshaffer please update

Comment (bshaffer) [reply gapic-generator/templates/default/service/client/_client.erb]: Done!

**Review by dazuma — APPROVED** [review]
Thanks Brent!

**Review by viacheslav-rostovtsev — APPROVED** [review]
LGTM

**Pull request merged**